mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksLifecycleAwareFlow.kt

```diff
@@ -83,14 +83,22 @@ private fun startedChannel(owner: Lifecycle): Channel<Boolean> {
     return channel
 }
 
-private inline fun <T : Any?> SelectBuilder<Unit>.onReceive(
+private inline fun <T : Any?, R : Any?> SelectBuilder<SelectResult<R>?>.onReceive(
     channel: ReceiveChannel<T>,
     crossinline onClosed: () -> Unit,
-    noinline onReceive: suspend (value: T) -> Unit
+    noinline onReceive: suspend (value: T) -> SelectResult<R>?
 ) {
     channel.onReceiveCatching {
         val result = it.getOrNull()
-        if (result === null) onClosed()
-        else onReceive(result)
+        if (result === null) {
+            onClosed()
+            null
+        } else {
+            onReceive(result)
+        }
     }
 }
+
+private data class SelectResult<T : Any?>(val payload: T)
+
+private data class FlowResult<T : Any?>(val payload: T)
```

Comment:
For performance reasons it makes sense to switch to a token (see how coroutines team [uses](https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/operators/Transform.kt#LL126C24-L126C24) it for example)

In this case it will be:
```
val NULL = Any()

var flowResult: Any? = NULL
var isClosed = false

val result = select {
    onReceive(startedChannel...) { value -> 
        started = value
        val result = flowResult
        if (started && result != NULL) {
           result
        } else { 
           NULL
        }
    }
    onReceive(flowChannel....) { value ->
       flowResult = value
        if (started && result != NULL) {
           flowResult
        } else { 
           NULL
        }
      }
if (result != NULL) {
   emit(result as T?)
}       
```

Or at least it doesn't need to be `data` class

Reply:
Thanks for comment! 
I applied this in [5425029](https://github.com/airbnb/mavericks/pull/678/commits/5425029ed54890d9604b99f29c01e7e67550d140) 👍